src/Analysis/Ast/Impl/Types/PythonClassType.cs

```diff
@@ -281,5 +284,55 @@ internal PythonClassType(string name, Location location)
         public bool Equals(IPythonClassType other)
             => Name == other?.Name && DeclaringModule.Equals(other?.DeclaringModule);
 
+        private IEnumerable<IPythonType> DisambiguateBases(IEnumerable<IPythonType> bases, IExpressionEvaluator eval) {
+            if (bases == null) {
+                return Array.Empty<IPythonType>();
+            }
+
+            if (eval == null) {
+                return FilterCircularBases(bases).Where(b => !b.IsUnknown());
+            }
+
+            var newBases = new List<IPythonType>();
+            foreach (var b in bases) {
+                var imported = eval.LookupImportedNameInScopes(b.Name, out _);
```

Comment:
Does it matter in which scope we have found the import?

Reply:
The search try to see if there is something imported before it is being declared. Outer scopes matter, i.e.
```python
from X import A

def func():
    class A(A): ...
    return A()
```
should find imported `A` up above and resolve base to it.
  
```

Reply:
What about the opposite case:
```python
class A(A): ...

def func():
    from X import A
    return A()
```

Reply:
```cs
class A(A): ...

def func():
    from Base import A
    return A()

x = func()
";
            var analysis = await GetAnalysisAsync(code);
            analysis.Should().HaveVariable("x").OfType("A").Which.Should().HaveMember("methodABase");
```
passes